src/coreclr/src/jit/morph.cpp

```diff
@@ -13757,6 +13757,13 @@ GenTree* Compiler::fgMorphSmpOpOptional(GenTreeOp* tree)
     switch (oper)
     {
         case GT_ASG:
+            // Make sure we're allowed to do this.
+            if (optValnumCSE_phase)
+            {
+                // It is not safe to reorder/delete CSE's
+                break;
+            }
+
```

Comment:
How does reordering this change the behavior here?

It seems to me that now we will break here, but before we would have also always done a break

Reply:
GitHub doesn't show you the real change, 
![image](https://user-images.githubusercontent.com/22642771/77206173-9766ac00-6ab3-11ea-9922-dd5a8c5fc1ac.png)